lib/Sema/CSSimplify.cpp

```diff
@@ -7852,8 +7869,9 @@ bool ConstraintSystem::recordFix(ConstraintFix *fix, unsigned impact) {
 
   // Record the fix.
 
-  // If this is just a warning it's shouldn't affect the solver.
-  if (!fix->isWarning()) {
+  // If this has zero impact or is just a warning, it shouldn't affect the
+  // solver.
+  if (!fix->isWarning() && impact > 0) {
```

Comment:
Is `impact > 0` used here to avoid `worseThanBestSolution()` check? This condition seems strange because impact of `0` doesn't change anything which means that the solution was already `worse`, so instead of failing right away we'd just fail a bit later after simplifying  one of the active constraints...

Reply:
@xedin Ah okay, I was under the impression that because we're not touching the score, there wouldn't be much point in checking `worseThanBestSolution`, but I guess there are other cases where we're touching the score and not checking it. Should we also be checking `worseThanBestSolution` for when the fix is a warning then?

Reply:
I think we should just ways do that and reset `impact` for warning to be `0` because I believe it was always `1` before that's why that check exists.

Reply:
Or rather call the `increaseScore` only if it's not a warning maybe.

Reply:
@xedin Yeah, I was thinking we could just move the call to `worseThanBestSolution` to just below this if statement. Checking `impact > 0` is still useful as it'll avoid us printing the `(increasing score due to ` in the diagnostics dump (though we could move the check into `increaseScore`).

Reply:
I think we should just move that check to the `increaseScore` instead, seems like a better place for it.

Reply:
@xedin Thanks, I've gone ahead and made that change. Does it look good?

Reply:
Looks great, thank you!